src/graph/input.rs

```diff
@@ -74,19 +86,18 @@ impl GraphWitness {
         serde_json::to_writer(std::fs::File::create(path)?, &self).map_err(|e| e.into())
     }
 }
-
 /// The input tensor data and shape, and output data for the computational graph (model) as floats.
 /// For example, the input might be the image data for a neural network, and the output class scores.
 #[derive(Clone, Debug, Deserialize, Default, Serialize)]
 pub struct GraphInput {
     /// Inputs to the model / computational graph (can be empty vectors if inputs are coming from on-chain).
     /// TODO: Add retrieve from on-chain functionality
-    pub input_data: Vec<Vec<f32>>,
+    pub input_data: DataSource,
```

Comment:
we def want a deserializer to that we can retain simple input.json that is either vector of floats or some on-chain calldata. OR how are you envisioning folks using this ? Is it either a vector of f32 OR a bunch of calldata ?

Reply:
Vector of f32 or a bunch of calldata at the moment.